### PR TITLE
[Improvement]: Introduce TableRuntimeFactory interface

### PR DESCRIPTION
**Pull request by baiyangtx**
## Brief change log

- Introduce the TableRuntimeFactory interface with plugin api.
- Refactor code about how TableRuntime is created.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: dist/src/main/amoro-bin/conf/plugins/table-runtime-factories.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This file is config file for plugin metric reporter
+# configurations of metric reporters
+
+table-runtime-factories:
+  - name: default            # configs for prometheus exporter
```
Comment (Aireed) [inline dist/src/main/amoro-bin/conf/plugins/table-runtime-factories.yaml]: the comment is incorrent

> # configs for prometheus exporter

File: amoro-ams/src/main/java/org/apache/amoro/server/table/DefaultTableService.java [hunk 2]
```diff
@@ -170,20 +172,29 @@ public void initialize() {
...
-                      identifier, tableRuntimeMeta, DefaultTableRuntime.REQUIRED_STATES, states);
-              DefaultTableRuntime tableRuntime = new DefaultTableRuntime(store, this);
-              tableRuntimeMap.put(tableRuntimeMeta.getTableId(), tableRuntime);
-              tableRuntime.registerMetric(MetricManager.getInstance().getGlobalRegistry());
-              tableRuntimes.add(tableRuntime);
-            });
+
+    for (TableRuntimeMeta tableRuntimeMeta : tableRuntimeMetaList) {
+      ServerTableIdentifier identifier = identifierMap.get(tableRuntimeMeta.getTableId());
+      if (!identifierMap.containsKey(tableRuntimeMeta.getTableId())) {
```
Comment (Aireed) [inline amoro-ams/src/main/java/org/apache/amoro/server/table/DefaultTableService.java]: why not check identifier ?
```suggestion
      if (identifier != null) {
```

**Comment by baiyangtx** [comment]
> I left some tiny comment. PTAL

Thx for your review, by the way, this PR using plugin system to load TableRuntimeFactory, and I found the plugins config is in `/conf/plugins `  dir.

Could we merge plugin configs to `/conf/config.yaml` ? 

WDYT? @zhoujinsong @Aireed

**Comment by Aireed** [comment]
> > I left some tiny comment. PTAL
> 
> Thx for your review, by the way, this PR using plugin system to load TableRuntimeFactory, and I found the plugins config is in `/conf/plugins ` dir.
> 
> Could we merge plugin configs to `/conf/config.yaml` ?
> 
> WDYT? @zhoujinsong @Aireed

From a functional perspective, they can be merged. +1. waiting for others to feedback

**Review by Aireed — APPROVED** [review]
LGTM. thanks for your work
